### PR TITLE
TDS-1162: Feature/delete assessment

### PR DESCRIPTION
**Pull request by jeffjohnson9046**
[TDS-1162](https://jira.fairwaytech.com/browse/TDS-1162):  Add code to delete assessments from the TIS.

*NOTES*
* There are a large number of file changes due to different versions of Visual Studio, updating location of dependencies and adding a new project to house unit tests.
* I'm adding @jtreuting as a reviewer because he has some experience working with TIS and .NET in general.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,12 @@ syntax: glob
 *.userosscache
 *.sln.docstates
 
+# Mac/OS X-specific files
+.DS_Store
+
+# Visual Studio 2015 cache/options directory
+.vs/
+
```
Comment (jeffjohnson9046) [inline .gitignore:16]: Added some additional directories that are created by OS X (when working via parallels) and the newer versions of Visual Studio

File: Common/TDSQASystemAPI/TDSQASystemAPI.csproj [hunk 2]
```diff
@@ -114,12 +114,24 @@
     <Reference Include="System.EnterpriseServices">
       <RequiredTargetFramework>4.5</RequiredTargetFramework>
     </Reference>
-    <Reference Include="System.Net.Http" />
-    <Reference Include="System.Net.Http.Formatting, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+    <Reference Include="System.Net" />
+    <Reference Include="System.Net.Http">
+      <HintPath>..\..\TISServices\TISServices\packages\System.Net.Http.2.0.20126.16343\lib\net40\System.Net.Http.dll</HintPath>
```
Comment (jeffjohnson9046) [inline Common/TDSQASystemAPI/TDSQASystemAPI.csproj:119]: Added missing dependencies that were not previously captured by NuGet (which is a package manager program for .NET).

File: Common/TDSQASystemAPI/app.config [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<configuration>
```
Comment (jeffjohnson9046) [inline Common/TDSQASystemAPI/app.config:2]: Created as a result of adding/updating dependencies.

File: TDSQAService/OSS.TIS/SQL/TISDB/3_ScoringDaemonConfiguration.sql [hunk 4]
```diff
@@ -8,9 +8,9 @@ declare @dbname varchar(255)
 declare @scoringDaemonName varchar(50)
 
 -- these can be IP or server name
-set @IP = 'my.tisdb.org'
-set @privateIP = 'my.tisdb.org' -- can be different if you want to use a private IP 
-set @dbname = 'MY_TIS_DB'
+set @IP = 'my.tisdb.org' -- Database Host Name or IP
+set @privateIP = NULL -- can be different if you want to use a private IP 
+set @dbname = 'OSS_TIS'
 set @scoringDaemonName = 'MyTISScoringDaemon'
```
Comment (jeffjohnson9046) [inline TDSQAService/OSS.TIS/SQL/TISDB/3_ScoringDaemonConfiguration.sql]: Left this name in place; we instruct the user to create a db named `OSS_TIS`.  If the user still wants to change it, tehy can.

File: TISServices/TISServices/TISServices.sln [hunk 5]
```diff
@@ -1,6 +1,8 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 2012
+# Visual Studio 15
```
Comment (jeffjohnson9046) [inline TISServices/TISServices/TISServices.sln:3]: This file was modified as a result of using a newer version of Visual Studio and adding the new unit test project to the solution

File: TISServices/TISServices/Web.config [hunk 6]
```diff
@@ -15,14 +15,15 @@
...
+    <add name="itembank" connectionString="Data Source=[DATABASE IP OR URL];Initial Catalog=OSS_Itembank;User id=[USERNAME];Password=[PASSWORD]" providerName="System.Data.SqlClient" />
+    <add name="TDSQC" connectionString="Data Source=[DATABASE IP OR URL];Initial Catalog=OSS_TIS;User id=[USERNAME];Password=[PASSWORD]" providerName="System.Data.SqlClient" />
   </connectionStrings>
   
   <appSettings>
-    <add key="LogFilePath" value="D:\MyFileServer\Logs\TISServicesLog.txt"/>
-    <add key="AuthTokenCache:MaxSize" value="100"/>
-    <add key="AuthTokenCache:PurgeCount" value="9"/>
-    <add key="AuthTokenCache:SlidingExpirationMinutes" value="240"/>
+    <add key="LogFilePath" value="[DRIVE]:\path\to\logs\TISServicesLog.txt" />
```
Comment (jeffjohnson9046) [inline TISServices/TISServices/Web.config:23]: Made this more explicit that a user/developer is expected to replace this value.

File: TISServices/TISServices/TISServices.csproj.user [hunk 7]
```diff
@@ -1,5 +1,14 @@
...
+    <Use64BitIISExpress />
+    <IISExpressSSLPort />
+    <IISExpressAnonymousAuthentication />
+    <IISExpressWindowsAuthentication />
+    <IISExpressUseClassicPipelineMode />
+    <UseGlobalApplicationHostFile />
+  </PropertyGroup>
   <ProjectExtensions>
     <VisualStudio>
       <FlavorProperties GUID="{349c5851-65df-11da-9384-00065b846f21}">
```
Comment (trapnine) [inline TISServices/TISServices/TISServices.csproj.user]: Generally .user files are not checked in. Do we have to do this with our setup?

File: TISServices/TISUnitTests/AssessmentControllerIntegrationTest.cs [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+                {
+                    connection.Open();
+                    for (var i = 0; i < validationSqls.Length; i++)
+                    {
+                        using (var validationSqlCommand = new SqlCommand(validationSqls[i], connection))
+                        {
+                            validationSqlCommand.CommandType = CommandType.Text;
+                            int result = (int)validationSqlCommand.ExecuteScalar();
+
+                            Assert.AreEqual(0, result);
```
Comment (trapnine) [inline TISServices/TISUnitTests/AssessmentControllerIntegrationTest.cs]: This is good, but we need a test that confirms the delete method doesn't accidentally delete other tests. For example, make a test that loads two, and only deletes one test, then makes sure one is still there. (then this test can clean it all up)

Also, maybe I'm missing it, but do you check there is a matching tests to start with before attempting the delete anf confirming it's 0? Maybe the load failed or loaded somewhere else, but this test will still pass.

**Review by trapnine — CHANGES_REQUESTED** [review]
Looks good! Integration test needs beefing up a bit and we're committing a .user file.

File: TISServices/TISServices/TISServices.csproj.user (hunk 7 above)
Comment (jeffjohnson9046) [reply TISServices/TISServices/TISServices.csproj.user]: To the best of my knowledge we don't have to include it.  I'm a bit surprised this file got checked in, considering there's a `*.user` in the `.gitignore`.

File: TISServices/TISUnitTests/AssessmentControllerIntegrationTest.cs (hunk 8 above)
Comment (jeffjohnson9046) [reply TISServices/TISUnitTests/AssessmentControllerIntegrationTest.cs]: I've updated the integration test(s):
* Now two assessments are loaded into the database before each test executes
* The `ShouldDeleteTheSpecifiedAssessment()` test will delete one of the assessments (the grade 3 MATH assessment) and verify the other one (the grade 11 ELA assessment) still exists
* Added another integration test that will delete the remaining assessment

While I was working on this, I realized the integration tests depended on running in a particular order, so I fixed that too.

**Review by trapnine — APPROVED** [review]
Thanks for the changes.

**Pull request merged**